server/schema/order/res.js

```diff
@@ -39,6 +43,28 @@ function getModifiersPrice(modifiers, modifierIds) {
     throw new Error(error.message)
   }
 }
+async function findNearestStore(deliveryAddress) {
+  let stores = await Store.findAll()
+  let storeAddresses = getStoreAddresses(stores)
+  let url =
+    'https://maps.googleapis.com/maps/api/distancematrix/json?origins=' + deliveryAddress + '&destinations=' + storeAddresses + '&key=' + apiKey
+  let distances = await fetch(url).then(async res => await res.json())
+  let storeName = findNearestStoreName(distances)
+  return stores.find(store => store.get('gmapAddress') === storeName).get('id')
```

Comment:
_d.find(stores,['gmapAddress',storeName]).get('id')

Reply:
stores is a sequelize object, can not map it http://prntscr.com/m9694n

Reply:
Addition, using map then find will loop array twice times(performance isn't good)